pallets/collection-data-feed/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use cfg_traits::data::{DataCollection, DataRegistry};
+	use frame_support::{pallet_prelude::*, storage::bounded_btree_map::BoundedBTreeMap};
+	use orml_traits::{DataProviderExtended, OnNewData};
+	use sp_runtime::{
+		traits::{EnsureAddAssign, EnsureSubAssign},
+		DispatchError,
+	};
+
+	type DataValueOf<T> = Option<(<T as Config>::Data, <T as Config>::Moment)>;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// A data identification
+		type DataId: Parameter + MaxEncodedLen + Ord;
+
+		/// A collection identification
+		type CollectionId: Parameter + MaxEncodedLen + Ord;
+
+		/// Represents a data
+		type Data: Parameter + MaxEncodedLen + Ord;
+
+		/// Represents a timestamp
+		type Moment: Parameter + MaxEncodedLen;
```

Comment:
Not sure if we want to tie a lot of pallets with `cfg_primitives` only because of a `Moment` type. If, in the future, any other project wants to use any of our pallets, it can be more scalable if we let the user choose the same `Moment` type they use in their chains (that should match their `pallet_timestamp` configurations). 

Anyways, I opened an issue regarding this: https://github.com/centrifuge/centrifuge-chain/issues/1277 maybe we should decide on one option and be consistent with it.

Reply:
Yes, you are right. If anyone ever uses this externally would be annyoing to have a fixed moment.